qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java

```diff
@@ -337,9 +220,27 @@ protected void onCreate(Bundle savedInstanceState) {
         appBarMain = findViewById(R.id.app_bap_main);
         fab = (FloatingActionButton) findViewById(R.id.fab);
 
-        self = this;
-        contacts = new HashMap<>();
-        identities = new Identities();
+        Intent serviceIntent = new Intent(this, LocalQabelService.class);
+        bindService(serviceIntent, new ServiceConnection() {
+
+            @Override
+            public void onServiceConnected(ComponentName name, IBinder service) {
+                LocalQabelService.LocalBinder binder = (LocalQabelService.LocalBinder) service;
+                mService = binder.getService();
+                onLocalServiceConnected();
+            }
+
+            @Override
+            public void onServiceDisconnected(ComponentName name) {
+                mService = null;
+            }
+
+        }, Context.BIND_AUTO_CREATE);
+
+    }
+
+    private void onLocalServiceConnected() {
```

Comment:
I would move everything that does not depend on the LocalService out of this method and back into onCreate(). This should be at least (without guarantee) ```self = this```, some of the init methods, the backStackChangeListener etc.